### ISSUE TITLE
docker-api conflict when using docker cookbook

### ISSUE THREAD

**Issue by qubitrenegade** [chef/train#108]
Hello,

I am attempting to use the [docker cookbook](https://github.com/chef-cookbooks/docker), while using the Chef [audit cookbook](https://github.com/chef-cookbooks/audit) (actually, I'm trying to use the [Swarm cookbook](https://github.com/tacchino/swarm) which depends on the docker cookbook, I believe it's a compatability issue between the docker cookbook and the r-train) I get the error:

`Unable to activate r-train-0.12.0, because docker-api-1.28.0 conflicts with docker-api (~> 1.26.2)`

This appears to be because docker has a requirement for docker-api-1.28.0, while r-train has a dependency on `docker-api ~> 1.26.2`?

Is there a way to work around this other than disabling Compliance?

Thanks!

Full output of error message:

```
================================================================================
Error executing action `fetch` on resource 'compliance_profile[ssh]'
================================================================================

Gem::ConflictError
------------------
Unable to activate r-train-0.12.0, because docker-api-1.28.0 conflicts with docker-api (~> 1.26.2)

Cookbook Trace:
---------------
/var/chef/cache/cookbooks/audit/libraries/profile.rb:37:in `block (2 levels) in <class:ComplianceProfile>'
/var/chef/cache/cookbooks/audit/libraries/profile.rb:30:in `block in <class:ComplianceProfile>'
/var/chef/cache/cookbooks/compat_resource/files/lib/chef_compat/monkeypatches/chef/runner.rb:41:in `run_action'

Resource Declaration:
---------------------
# In /var/chef/cache/cookbooks/audit/recipes/default.rb

 30:   compliance_profile p do
 31:     owner o
 32:     server server
 33:     token token
 34:     inspec_version node['audit']['inspec_version']
 35:     action [:fetch, :execute]
 36:   end
 37: end

Compiled Resource:
------------------
# Declared in /var/chef/cache/cookbooks/audit/recipes/default.rb:30:in `block in from_file'

compliance_profile("ssh") do
  action [:fetch, :execute]
  retries 0
  retry_delay 2
  default_guard_interpreter :default
  declared_type :compliance_profile
  cookbook_name "audit"
  recipe_name "default"
  owner "base"
  inspec_version "0.22.1"
end

Platform:
---------
x86_64-linux
```

**Comment by jeremymv2** [comment]
@chris-rock can the r-train runtime dependency `docker-api ~> 1.26.2` be loosened to `~> 1.26` perhaps?

**Comment by chris-rock** [comment]
Thanks @qubitrenegade and @jeremymv2 for the report. We released train 0.12.1 that includes the fix proposed by @jeremymv2
